### ISSUE TITLE
[Question/Request] Loading Choices Without Reaching End

### ISSUE THREAD

**Issue by lilytastic** [inkle/ink-unity-integration#29]
The way Ink is set up, choices are always chunked together at the end of a story block. I want to be able to go through the story line by line, and show and hide choices at various points, so that certain ones will appear to be 'time-limited'. It's a technique featured in Ladykiller in a Bind, and I thought it added a whole new dimension to the CYOA style.
Unfortunately, I can't seem to get any choices without reaching the end of the story block. Right now I'm able to accomplish this by using Continue() to reach the end of a block, storing each line in a separate List, getting the choices at the end, and then feeding the player each line from my own List with the choices all stored and ready to go. Actually deciding when certain choices appear or disappear is handled with tags. Here's what that looks like: https://dl.dropboxusercontent.com/u/17586791/Screen%20Shot%202017-02-17%20at%204.02.42%20PM.png
The problem is that every bit of logic in those lines is triggered. I'd love to be able to skip lines by letting the player select a choice before the end of the block, where it would normally appear. 

tl;dr: Is there a way to get all the upcoming choices without reaching the end of a story block, or is there a way to Continue() without triggering any logic, or storing logic for later, or any other method that would let me store choices without having to trigger logic on every line?

I realize this is kind of a weird request so I wouldn't be surprised if it isn't possible.

**Comment by joethephish** [comment]
Here's one idea, but it's not particularly pretty. The crux is that you try to work with ink's step-by-step flow so that you maintain ink's logic in sync with what it expects.

In order to do that, that this does is explicitly offers the same set of choices at each step, and then has a special "NONE" choice that you should automatically have the game silently pick as you step through the story line by line.

```
LIST ChoiceNumbers = one, two, three, four, five, six

VAR c = ()

"Wake him."

You consciousness comes back.
~ c += one
-> choices ->

A Delgoren guard blocks the lone light.
~ c -= one
-> choices ->

The officer looks down at you.
~ c += (one, two, four)
-> choices ->

You're supposed to be dead.
~ c += three
-> choices ->

You cough up a mouthful of water.
-> choices ->


= choices
+ {c ? one} "Augh!"
+ {c ? two} "That wasn't necessary"
+ {c ? three} [Remain silent]
+ {c ? four} "...What?"[]
+ {c ? five} "Who are you?"[]
+ [NONE] 
    ->->
- -> END
```

Or, another simpler / easier to understand version that would require listing the name of all the choices (though you could potentially group them):

```
"Wake him."

You consciousness comes back.
<- augh
+ [NONE]
-

A Delgoren guard blocks the lone light.


The officer looks down at you.
<- augh
<- not_necess
<- what
+ [NONE]
-

You're supposed to be dead.
<- augh
<- not_necess
<- silent
<- what
+ [NONE]
-

You cough up a mouthful of water.
<- augh
<- not_necess
<- silent
<- what
+ [NONE]
-

= augh
+ "Augh!"

= not_necess
+ "That wasn't necessary"

= silent
+ [Remain silent]

= what
+ "...What?"[]

= who
+ "Who are you?"[]
```

Those are the best ideas I can come up with off the top of my head!

**Comment by lilytastic** [comment]
That's not too bad. Thanks!
